### PR TITLE
Improve build-api-based-message-extension.md

### PR DESCRIPTION
**Pull request by Benjiiim**
Add more details and resources.

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/014a24fa0d4c8af90e4ff1ba527698d5e3d76eaa/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 60 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/ce042b10-489f-4c4e-ab62-ba6516c94e0b) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/cbcb56e7283a5b48a2cfdb733b99baace6384a2d/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 60 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/76cc07ae-fd79-48ca-8df9-da59c59d501e) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 1]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
   > [!NOTE]
   > The API receives a Microsoft Entra token with the scope set to `access_as_user` as registered in the Azure portal. However, the token isn't authorized to call any other downstream APIs, such as Microsoft Graph.
 
+For more information about validating access token, see [Validate tokens](/azure/active-directory/develop/access-tokens#validate-tokens).
```
Comment (v-ypalikila) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:956]:
```suggestion
For more information on how to validate an access token, see [Validate tokens](/azure/active-directory/develop/access-tokens#validate-tokens).
```

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 2]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
...
+There are several libraries to handle JWT validation. For basic validation, ensure to check the following:
+
+* The token is well-formed.
+* The token was issued by the intended authority.
+* The token is targeted to the web API.
+
+Ensure the following when validating the token:
+
+* Valid SSO tokens are issued by Microsoft Entra ID. The `iss` claim in the token must start with this value.
+* The token's `aud1` parameter is set to the app ID generated during Microsoft Entra app registration.
```
Comment (v-ypalikila) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:967]:
```suggestion
* The `aud1` claim in the token should match the app ID from Microsoft Entra app registration.
```

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 3]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
   > [!NOTE]
   > The API receives a Microsoft Entra token with the scope set to `access_as_user` as registered in the Azure portal. However, the token isn't authorized to call any other downstream APIs, such as Microsoft Graph.
 
+For more information about validating access token, see [Validate tokens](/azure/active-directory/develop/access-tokens#validate-tokens).
+
+There are several libraries to handle JWT validation. For basic validation, ensure to check the following:
```
Comment (v-ypalikila) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:958]:
```suggestion
There are several libraries to validate JSON Web Token (JWT). For basic validation, ensure that you check the following:
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/7971cd343bcd22c10c6f83f980332f4ffa2704b3/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 58 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/5b02eae9-66ed-49b9-abf4-cee3f96f2392) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 4]
```diff
@@ -830,7 +830,7 @@ Update the following properties in the app manifest file:
 
    &nbsp;&nbsp;:::image type="content" source="../assets/images/authentication/teams-sso-tabs/sso-manifest.png" alt-text="Screenshot shows the app manifest configuration.":::
 
-* `microsoftEntraConfiguration`: Enables Single sign-on authentication for your app. Configure the `supportsSingleSignOn` property to `true` to support SSO and  reduce the need for multiple authentications.
+* `authorization.microsoftEntraConfiguration`: Enables Single sign-on authentication for your message extension. Configure the `supportsSingleSignOn` property to `true` to support SSO and  reduce the need for multiple authentications. For more information, see [composeextensions](../resources/schema/manifest-schema.md#composeextensions).
```
Comment (v-kushals) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:833]:
```suggestion
* `authorization.microsoftEntraConfiguration`: Enables single sign-on (SSO) authentication for your message extension. Configure the `supportsSingleSignOn` property to `true` to support SSO and reduce the need for multiple authentications. For more information, see [composeExtensions](../resources/schema/manifest-schema.md#composeextensions).
```

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 5]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
   > [!NOTE]
   > The API receives a Microsoft Entra token with the scope set to `access_as_user` as registered in the Azure portal. However, the token isn't authorized to call any other downstream APIs, such as Microsoft Graph.
 
+For more information on how to validate an access token, see [Validate tokens](/azure/active-directory/develop/access-tokens#validate-tokens).
```
Comment (v-kushals) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:956]:
```suggestion
For more information on how to validate an access token, see [validate tokens](/azure/active-directory/develop/access-tokens#validate-tokens).
```

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 6]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
...
+
+There are several libraries to validate JSON Web Token (JWT). For basic validation, ensure that you check the following:
```
Comment (v-kushals) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:958]:
```suggestion
There are several libraries to validate JSON Web Tokens (JWT). For basic validation, ensure that you check the following:
```

Comment (v-kushals) [reply msteams-platform/messaging-extensions/build-api-based-message-extension.md:958]: @Benjiiim, if there is just one JSON Web Token, change it to "to validate a JSON Web Token (JWT)".

File: msteams-platform/messaging-extensions/build-api-based-message-extension.md [hunk 7]
```diff
@@ -957,6 +953,20 @@ After the API-based message extension gets a request header with token, perform
...
+
+* The token is well-formed.
+* The token was issued by the intended authority.
+* The token is targeted to the web API.
+
+Ensure the following when validating the token:
```
Comment (v-kushals) [inline msteams-platform/messaging-extensions/build-api-based-message-extension.md:964]:
```suggestion
When validating the token, ensure the following:
```

**Review by v-kushals — CHANGES_REQUESTED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/89f05cac8585ac19cf313d5673b91c4a4136a77f/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 57 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/65f87bf9-f286-447d-846b-e04efd76a2fd) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by v-kushals — APPROVED** [review]

**Review by v-deabira — APPROVED** [review]

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/5b295c3058403a1310e0c904f5a9f881cf940279/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 57 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/977825af-0cda-435e-9731-8384c880efb0) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/build-api-based-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/709e2906ec677c6d36b008990ec4243b94f4d8ed/msteams-platform%2Fmessaging-extensions%2Fbuild-api-based-message-extension.md) | 91 | 57 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/c6b10e58-0d14-4e9e-9625-17cf466e5ee4) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Pull request merged**